awx/main/dispatch/__init__.py

```diff
@@ -68,10 +77,23 @@ def pg_bus_conn(new_connection=False):
         conn.set_session(autocommit=True)
     else:
         if pg_connection.connection is None:
-            pg_connection.connect()
-        if pg_connection.connection is None:
-            raise RuntimeError('Unexpectedly could not connect to postgres for pg_notify actions')
-        conn = pg_connection.connection
+            for retry_count in range(MAX_RETRIES):
+                try:
+                    pg_connection.connect()
+                    conn = pg_connection.connection
+                except (psycopg2.OperationalError, OperationalError, Error) as reconn_error:
+                    logger.warning(f'Database unavailable. Retry with new connection in next {POLL_SECONDS} seconds. Attempt {retry_count}/{MAX_RETRIES}.')
+                    time.sleep(POLL_SECONDS)
+                    retry_conn = True
+                else: 
+                    if retry_conn:
+                        logger.warning('Run dispatcher restart due to database connection loss and restore.')
```

Comment:
logging nit-pick, this is used more broadly than the "dispatcher" for our task system. For the most part, this will apply to pg_notify listeners.

Reply:
This logging e ter only in case of connection fail.
This seems to happen only for the Main queue when postgres go down